### PR TITLE
Fixed user export scope to acknowledge FMCS

### PR DESCRIPTION
**Pull request by Godmartinz**
# Description

This changes the scope of the export to acknowledge Full Multi Company Support. So the requester of the export will only see users from their company if FMCS is enabled. This duplicates a lot of lines in order to work 😬 If there is a better way, I am all ears and fingers.

Fixes #[sc-25183]

## Type of change

Please delete options that are not relevant.

- [ x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* PHP version:
* MySQL version
* Webserver version
* OS version

# Checklist:

- [ ] I have read the Contributing documentation available here: https://snipe-it.readme.io/docs/contributing-overview
- [ ] I have formatted this PR according to the project guidelines: https://snipe-it.readme.io/docs/contributing-overview#pull-request-guidelines
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/Http/Controllers/Users/UsersController.php [hunk 1]
```diff
@@ -541,11 +543,74 @@ public function getExportUserCsv()
...
-
         $response = new StreamedResponse(function () {
+        $user = Auth::user();
+            $settings = Setting::getSettings();
             // Open output stream
             $handle = fopen('php://output', 'w');
 
+            if($settings->full_multiple_companies_support){
+                User::with('assets', 'accessories', 'consumables', 'department', 'licenses', 'manager', 'groups', 'userloc', 'company')
+                    ->where('company_id', '=', $user->company_id)
```
Comment (snipe) [inline app/Http/Controllers/Users/UsersController.php]: Hm, we should probably respect the superadmin permissions here though. If I'm a superadmin, I should be able to download all, no?

**Comment by Godmartinz** [comment]
added a `!$user ->isSuperUser()` to the if statement @snipe

File: app/Http/Controllers/Users/UsersController.php [hunk 2]
```diff
@@ -33,6 +33,8 @@
  */
 class UsersController extends Controller
 {
+    private $user;
```
Comment (marcusmoore) [inline app/Http/Controllers/Users/UsersController.php]: This doesn't look like it's needed?

File: app/Http/Controllers/Users/UsersController.php [hunk 3]
```diff
@@ -541,11 +543,74 @@ public function getExportUserCsv()
...
-
         $response = new StreamedResponse(function () {
+            $user = Auth::user();
+            $settings = Setting::getSettings();
             // Open output stream
             $handle = fopen('php://output', 'w');
 
+            if($settings->full_multiple_companies_support && !$user->isSuperUser() ){
+                User::with('assets', 'accessories', 'consumables', 'department', 'licenses', 'manager', 'groups', 'userloc', 'company')
+                    ->where('company_id', '=', $user->company_id)
```
Comment (marcusmoore) [inline app/Http/Controllers/Users/UsersController.php]: > This duplicates a lot of lines in order to work 😬 If there is a better way, I am all ears and fingers.

Is this line the only change between the two? If so you can start the query, do the `if` check and add it to the query there.

Something like:

```php
$query = User::with(...)

if($settings->full_multiple_companies_support && !$user->isSuperUser() ){
	$query = $query->where('company_id', '=', $user->company_id);
}

$query->orderBy->etc...
```

**Review by marcusmoore — COMMENTED** [review]
I didn't run the code locally but here's a note on reducing duplication.

File: app/Http/Controllers/Users/UsersController.php (hunk 3 above)
Comment (Godmartinz) [reply app/Http/Controllers/Users/UsersController.php]: TIL. very cool, updated the query.

**Review by marcusmoore — APPROVED** [review]
Pushed a formatting commit (with permission).

Looks good 😄

**Comment by snipe** [comment]
Your instincts are good here :) I think this will be covered with the other user scoping stuff I'm working on (and should hopefully have available later today), so I'll close it for now tho.

**Pull request closed without merging**